sklearn/ensemble/_iforest.py

```diff
@@ -327,6 +327,18 @@ def fit(self, X, y=None, sample_weight=None):
             check_input=False,
         )
 
+        self._average_path_length_max_samples = _average_path_length(
+            [self._max_samples]
+        )
```

Comment:
I am under the impression that we don't need to store this value but we can compute it outside of the for loop in `_compute_score_samples`

Reply:
We can, but every call to predict on the tree would calculate it again.

Reply:
Yes, but it is super cheap to compute. Looking at `_average_path_length`, it comes back to some almost scalar operation. It is around 30 microseconds on my machine. It is negligible in regards to the prediction time is x10,000 more important. It avoids storing an attribute also.

Reply:
Agreed. Updated